desktop/app/dockIcon.js

```diff
@@ -0,0 +1,19 @@
+import app from 'app'
+
+var visibleCount = 0
+
+export default function () {
+  if (++visibleCount === 1) {
+    app.dock.show()
+  }
+  let alreadyHidden = false
+  return () => {
+    if (alreadyHidden) {
+      throw new Error('Tried to hide the dock icon twice')
```

Comment:
Hopefully, or at least make Electron throw an error. Could just make this a noop instead, but I figure this is better to catch our errors.